### PR TITLE
Fix flakiness and race conditions in ProjectWorkflowServiceTest

### PR DESCRIPTION
**Pull request by j-roskopf**
### Summary

This PR resolves test flakiness and CI timeouts in `ProjectWorkflowServiceTest` (specifically `manualCompleteBuildAttemptAdvancesWorkflow`) and `AgentQueuedFollowUpTest`.

#### Root Causes & Fixes
1. **Terminal status monitor resurrection race**:
   - When `completeWorkflowRun(buildRun.id)` was called, it synchronously marked the task `status = Done` via `finishTask`, stamped `finishedAtMillis`, launched `reconcileWorkflowRun`, and initiated async terminal teardown (`terminals.stop(taskId)` on `Dispatchers.IO`).
   - While the terminal was tearing down, the background terminal status monitor collected a soft `Working` status snapshot. Because `applyStatusSnapshot` lacked guards against inactive completed workflow stages, it resurrected the already-finalized task back to `status = Working` and cleared `finishedAtMillis`.
   - Then `runProcess` saw `handle.stopRequested` or loop exit and invoked `finishTask` a second time, triggering a second `reconcileWorkflowRun` and creating a duplicate Review attempt (`attempts.size == 2`).
   - **Fix**: Guarded `applyStatusSnapshot` against resurrecting finalized inactive workflow tasks (`task.workflowStage != null && task.finishedAtMillis != null && !task.isActive`) or tasks with `stopRequested`, and guarded `runProcess` against calling `finishTask` multiple times when `stopRequested` is set.
2. **Review task active state polling race in test**:
   - In `manualCompleteBuildAttemptAdvancesWorkflow`, `WorkflowAdapter` sets `stageDelayMillis = 0` for review stages (`/bin/sh -c true`), meaning the review task transitions from `Working` (`isActive = true`) to `Done` (`isActive = false`) in ~1ms.
   - The test previously asserted `await { tasks.any { it.workflowStage == Review && it.isActive } }`. Under loaded CI where polling intervals exceeded the 1ms execution window, the condition was never observed, leading to a 15-minute test timeout (`TimeoutCancellationException`).
   - **Fix**: Changed the test to await the existence of the Review stage and then wait for the pair to complete, verifying the single review attempt and verdict deterministically.
3. **Out-of-order follow-up queueing race**:
   - `queueFollowUp` previously launched coroutines directly on `Dispatchers.IO` without serialization. When multiple follow-ups were queued sequentially (e.g. "second message" followed immediately by "third message"), the coroutines raced on the thread pool and could be enqueued out-of-order (`[third message, second message]`), causing the test's ordering assertion loop to time out.
   - **Fix**: Introduced a dedicated limited-parallelism dispatcher (`limitedParallelism(1)`) and a per-task mutex (`followUpMutex`) to serialize follow-up queueing in strict FIFO order.

### Verification
- Ran 10 consecutive iterations of `manualCompleteBuildAttemptAdvancesWorkflow` with `--rerun-tasks` (10/10 passed).
- Ran 5 consecutive iterations of `AgentQueuedFollowUpTest` with `CI=true` and `--rerun-tasks` (5/5 passed).
- Ran the full `ProjectWorkflowServiceTest` test suite (all 45 active tests passed).
- Verified edge cases (`invalidReviewContractsNeverAdvanceOrConsumeAReviewFailure`, `invalidVerifierOutputNeverCompletesBuildOrSchedulesAnotherStage`, `malformedReviewNeedsAttentionWithoutConsumingTheFailureLimit`, `lateReviewArtifactRecoveredOnResumeWithoutStartingAnotherReview`).
- Full CI matrix is green across all platforms (macOS, Linux, Windows desktop tests, screenshots, and builds).

### CODE REVIEW

File: data/agents/src/desktopMain/kotlin/app/andy/desktop/service/agents/DesktopAgentRunService.kt [hunk 1]
```diff
@@ -2547,6 +2547,14 @@ class DesktopAgentRunService(
             }
         } + extraProviderLaunchEnv(taskForLaunch)
 
+        if (handle.stopRequested || currentTask(taskId)?.finishedAtMillis != null) {
+            val current = currentTask(taskId)
+            if (current != null && current.finishedAtMillis == null) {
+                finishTask(taskId, AgentStatus.Done, exitCode = null, error = null, stoppedByUser = true)
+            }
+            return
```
[the remark this answers is not in the document]
Comment (j-roskopf) [reply data/agents/src/desktopMain/kotlin/app/andy/desktop/service/agents/DesktopAgentRunService.kt]: Fixed in commit 2b3c986b: Exempted `quietResume` from the finalized-task guards in `runProcess` so view-only terminal reattaches proceed through terminal launch.

File: data/agents/src/desktopMain/kotlin/app/andy/desktop/service/agents/DesktopAgentRunService.kt [hunk 2]
```diff
@@ -6514,6 +6535,10 @@ class DesktopAgentRunService(
 
     private fun applyStatusSnapshot(taskId: String, snapshot: AgentStatusSnapshot) {
         val task = currentTask(taskId) ?: return
+        if (handles[taskId]?.stopRequested == true) return
+        if (task.finishedAtMillis != null && !task.isActive && (snapshot.status == AgentStatus.Working || snapshot.status == AgentStatus.Blocked)) {
+            return
```
[the remark this answers is not in the document]
Comment (j-roskopf) [reply data/agents/src/desktopMain/kotlin/app/andy/desktop/service/agents/DesktopAgentRunService.kt:6547]: Fixed in commit 2b3c986b: Restricted the inactive/finished snapshot guard in `applyStatusSnapshot` strictly to project workflow stages (`task.workflowStage != null`), preserving normal live terminal Working transitions for interactive chat sessions.

File: data/agents/src/desktopMain/kotlin/app/andy/desktop/service/agents/AgentSessionAttention.kt [hunk 3]
```diff
@@ -71,10 +71,6 @@ internal fun shouldIgnoreStatusSnapshot(
     ) {
         return true
     }
-    // Live interactive sessions: always accept Working (soft or confident).
-    if (terminalLive && snapshot.status == AgentStatus.Working) {
-        return false
-    }
     // Soft Working after confident Done/Error is remount / boot noise.
     // Soft Working after Blocked is a real turn continuation — allow it.
     if (snapshot.status == AgentStatus.Working &&
```
[the remark this answers is not in the document]
Comment (j-roskopf) [reply data/agents/src/desktopMain/kotlin/app/andy/desktop/service/agents/AgentSessionAttention.kt:80]: Addressed in commit ef979870: Reverted `AgentSessionAttention.kt` to preserve the tested `terminalLive` precedence.
